app/src/main/java/io/github/hidroh/materialistic/CustomTabsDelegate.java

```diff
@@ -110,8 +112,36 @@ void onServiceDisconnected() {
     }
 
     private static String getPackageNameToUse(Context context) {
-        // packagesSupportingCustomTabs contains all apps that can handle both VIEW intents
-        // and service calls.
+        List<String> browserWithCustomTabsSupport = getBrowserWithCustomTabsSupport(context);
+        String defaultBrowser = getDefaultBrowser(context);
+
+        for (String browser : browserWithCustomTabsSupport) {
+            if (browser.equals(defaultBrowser)) {
+                return browser;
+            }
+        }
+
+        return browserWithCustomTabsSupport.get(0);
```

Comment:
We should still apply the fallback logic to return null if `browserWithCustomTabsSupport.isEmpty()` here, similar to old code.

Reply:
Done